README.md

```diff
@@ -168,33 +162,32 @@ class State(pc.State):
     """The app state."""
     prompt = ""
     image_url = ""
-    image_processing = False
-    image_made = False
+    processing = False
+    complete = False
 ```
 
 The state defines all the variables (called vars) in an app that can change and the functions that change them.
 
-Here the state is comprised of a `prompt` and `image_url`. There are also the booleans `image_processing` and `image_made` to indicate when to show the circular progress and image.
+Here the state is comprised of a `prompt` and `image_url`. There are also the booleans `processing` and `complete` to indicate when to show the circular progress and image.
 
 ### **Event Handlers**
 
 ```python
-    def process_image(self):
-        """Set the image processing flag to true and indicate image is not made yet."""
-        self.image_processing = True
-        self.image_made = False
-
-    def get_image(self):
-        """Get the image from the prompt."""
-        response = openai.Image.create(prompt=self.prompt, n=1, size="1024x1024")
-        self.image_url = response["data"][0]["url"]
-        self.image_processing = False
-        self.image_made = True
+def get_image(self):
+    """Get the image from the prompt."""
+    if self.prompt == "":
+        return pc.window_alert("Prompt Empty")
+
+    self.processing, self.complete = True, False
+    yield
+    response = openai.Image.create(prompt=self.prompt, n=1, size="1024x1024")
+    self.image_url = response["data"][0]["url"]
+    self.processing, self.complete = False, True
 ```
 
 Within the state, we define functions called event handlers that change the state vars. Event handlers are the way that we can modify the state in Pynecone. They can be called in response to user actions, such as clicking a button or typing in a text box. These actions are called events.
 
-Our DALL·E. app has two event handlers, `process_image` to indicate that the image is being generated, and `get_image`, which calls the OpenAI API.
+Our DALL·E. app has an event handlers, `get_image` to which get this image from the openai api. Using yield in the middle of an event handler will cause the UI to update. Otherwise the UI will update at the end of the event handler.
```

Comment:
"an event handlers" -> "an event handler". Also can we put `yield` in backticks and format it as "OpenAI API"